xarray/core/dataarray.py

```diff
@@ -2015,14 +2015,20 @@ def shift(self, **shifts):
         variable = self.variable.shift(**shifts)
         return self._replace(variable)
 
-    def roll(self, **shifts):
+    def roll(self, roll_coords=None, **shifts):
```

Comment:
It might be nice to use `utils.either_dict_or_kwargs` here, to handle potential (unlikely) conflicts between the name "roll_coords" and dimension names.

Example usage:
https://github.com/pydata/xarray/blob/ded0a684136540962bcc409e6272b1cebb5af30a/xarray/core/dataarray.py#L755

Reply:
Not too sure I understand this util; let me know if I implemented it inaccurately.